### PR TITLE
[Improvement](nereids) Make sure to catch and record exception for every materialization context

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

1. Make sure instance when change params of StructInfo,Predicates.
2. Catch and record exception for every materialization context, this make sure that if throw exception when one materialization context rewrite, it will not influence others.
3. Support to mv rewrite when hava count function when aggregate without group by

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/StructInfo.java [hunk 1]
```diff
@@ -216,11 +226,39 @@ public static List<StructInfo> of(Plan originalPlan) {
...
+        }
+        // collect struct info fromGraph
+        List<CatalogRelation> relations = new ArrayList<>();
+        Map<RelationId, StructInfoNode> relationIdStructInfoNodeMap = new HashMap<>();
+        Map<Expression, Expression> shuttledHashConjunctsToConjunctsMap = new HashMap<>();
+        Map<ExprId, Expression> namedExprIdAndExprMapping = new HashMap<>();
+        boolean valid = collectStructInfoFromGraph(hyperGraph, topPlan, shuttledHashConjunctsToConjunctsMap,
+                namedExprIdAndExprMapping,
+                relations,
+                relationIdStructInfoNodeMap);
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/StructInfo.java:256]: Maybe can use ImmutableList

**Review by keanji-x — APPROVED** [review]

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/StructInfo.java (hunk 1 above)
Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/StructInfo.java]: Have fixed it

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Comment by seawinde** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**